File: src/dotnet/HelloMutation.Domain/Entities/CartItem.cs
using System;

namespace HelloMutation.Domain.Entities
{
    public record CartItem
    {
        public CartItem(
            Book book,
            ushort quantity
        ) => (Book, Price, Quantity) = (book, book.GetPriceAt(DateTime.Now).Value, quantity);

        private Book _book;
        public Book Book
        {
            get => _book;
            init => _book = value ?? throw new ArgumentNullException(nameof(Book));
        }
        public decimal Price { get; }
        public ushort Quantity { get; }
    }
}

File: src/javascript/src/domain/entities/cart_item.ts
import { Book } from './book'

export class CartItem {
    private _book!: Book
    public get book(): Book {return this._book}

    private _price!: number
    public get price(): number {return this._price}

    private _quantity!: number
    public get quantity(): number {return this._quantity}

    constructor(init: Partial<CartItem>) {
        this._book = init.book!
        this._quantity = init.quantity!
        this._price = init.book!.getPriceAt(new Date())!.value ?? 0
    }
}